packages/ui/src/components/va-skeleton/VaSkeleton.vue

```diff
@@ -0,0 +1,210 @@
+<template>
+  <component class="va-skeleton" :class="classes" :is="tag">
+    <slot />
+    <div class="va-skeleton-wave" v-if="animation === 'wave'" />
+  </component>
+</template>
+
+<script lang="ts">
+import { defineComponent, onMounted, ref, computed, PropType } from 'vue'
+import { useBem, useColors } from '../../composables'
+
+export default defineComponent({
+  name: 'VaSkeleton',
+
+  inheritAttrs: true,
+
+  props: {
+    color: { type: String, default: 'backgroundElement' },
+    delay: { type: Number, default: 100 },
+
+    tag: { type: String, default: 'div' },
+    inline: { type: Boolean, default: false },
+
+    animation: { type: String as PropType<'pulse' | 'wave' | 'none'>, default: 'pulse' },
+
+    lines: { type: Number, default: 1 },
+    height: { type: [Number, String], default: '1em' },
+    width: { type: [String], default: '100%' },
+    lineGap: { type: String, default: '8px' },
+    textWidth: { type: [String], default: '75%' },
+
+    variant: { type: String as PropType<'text' | 'circle' | 'rounded' | 'squared'>, default: 'squared' },
+  },
+
+  emits: [/* Define emits here */],
+
+  setup (props, { attrs }) {
+    const doShow = ref(false)
+
+    onMounted(() => {
+      setTimeout(() => {
+        doShow.value = true
+      }, props.delay)
```

Comment:
Shouldn't we clear timeouts on unmounted hook?

Reply:
Probably not, will hold component garbage collection for a bit, but given you don't set delay too high - that shouldn't matter at all.